drivers/counter/counter_mchp_xec.c

```diff
@@ -46,7 +46,7 @@ struct counter_xec_data {
 };
 
 #define COUNTER_XEC_REG_BASE(_dev)			\
-	((BTMR_Type *)					\
+	((struct btmr_regs *)					\
 	 ((const struct counter_xec_config * const)	\
```

Comment:
need to confirm this change works for MEC15xx as well.

Reply:
The code works well on MEC15
BTMR_Type is defined like this in MEC15
```
typedef struct btmr_regs
{
        __IOM uint32_t CNT;     /*!< (@ 0x00000000) BTMR Count            */
        __IOM uint32_t PRLD;    /*!< (@ 0x00000004) BTMR Preload          */
        __IOM uint8_t  STS;     /*!< (@ 0x00000008) BTMR Status           */
        uint8_t RSVDC[3];
        __IOM uint8_t IEN;      /*!< (@ 0x0000000c) BTMR Interrupt Enable */
        uint8_t RSVDD[3];
        __IOM uint32_t CTRL;    /*!< (@ 0x00000010) BTMR Control          */
} BTMR_Type;
```
I tried to use typedef in soc/arm/microchip_mec/mec172x/reg/mec172x_ecia.h for MEC 17 soc/arm/microchip_mec/common/reg/mec_timers.h, but the check function says new typedef was not allowed. So I modify it in the driver code.